spec/jobs/stream_api_requests_to_big_query_job_spec.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe StreamAPIRequestsToBigQueryJob, type: :job do
+  subject(:job) { described_class.perform_now(request_data, response_data, status_code, created_at) }
+
+  let(:bigquery) { instance_double("bigquery") }
+  let(:dataset)  { instance_double("dataset") }
+  let(:table)    { instance_double("table", insert: nil) }
+  let(:status_code) { 200 }
+  let(:created_at) { Time.zone.now.to_s }
+  let(:lead_provider) { LeadProvider.find_by(name: "Ambition Institute") }
+
+  let(:request_data) do
+    {
+      "path" => "/api/v3/participants/npq",
+      "params" => {},
+      "body" => { "foo" => "bar" }.to_json,
+      "headers" => {
+        "HTTP_VERSION" => "HTTP/1.1",
+        "HTTP_HOST" => "localhost:3000",
+        "HTTP_USER_AGENT" => "PostmanRuntime/7.39.0",
+        "HTTP_ACCEPT" => "*/*",
+        "HTTP_ACCEPT_ENCODING" => "gzip, deflate, br",
+        "HTTP_AUTHORIZATION" => "Bearer ambition-token",
+        "HTTP_CONNECTION" => "keep-alive",
+        "QUERY_STRING" => "",
+      },
+      "method" => "GET",
+    }
+  end
+  let(:response_data) do
+    {
+      "headers" => { "Content-Type" => "application/json" },
+      "body" => {
+        "data" => [{
+          "id" => "123456",
+          "type" => "npq-participant",
+          "attributes" => {
+            "full_name": "Test Example",
+            "teacher_reference_number": "123",
+          },
+        }],
+      }.to_json,
+    }
+  end
+
+  before do
+    APIToken.create_with_known_token!("ambition-token", lead_provider:)
+
+    allow(Google::Cloud::Bigquery).to receive(:new).and_return(bigquery)
+    allow(bigquery).to receive(:dataset).and_return(dataset)
+  end
+
+  describe "#perform" do
+    context "when the BigQuery table exists" do
+      before do
+        allow(dataset).to receive(:table).and_return(table)
+      end
+
+      it "enqueues a job" do
+        expect {
+          described_class.perform_later
+        }.to have_enqueued_job
+      end
+
+      context "when API Request response is success" do
+        it "sends correct data to BigQuery with no `response_body`" do
```

Comment:
Just curious; why do we scrub the response body but not the request body? couldn't the request body have PII? 🤔

Reply:
Think there's no request body with a potential PII in any requests body params to the API.
We're just removing the response body for successful calls (keeping for error responses).. as success calls returns huge amount of data, also includes PII.